fix(c/driver_manager,rust/driver_manager): establish standard platform tuples (#3313)
apache/arrow-adbc: docs/source/format/driver_manifests.rst

needed to successfully load a driver manifest.  The other keys are optional, but provide useful metadata
about the driver.

> Platform-Tuples
> ^^^^^^^^^^^^^^^

Reviewer: ```suggestion
Platform Tuples
^^^^^^^^^^^^^^^
```

> Since the manifests will utilize platform tuples to specify the different systems

Reviewer: ```suggestion
Since the manifests use platform tuples to specify the different systems
```

that drivers might be built for, it is important to create a consistent way to name

> these tuples. Specifically, consistent naming for OS and Architecture combinations

Reviewer: ```suggestion
these tuples. Specifically, consistent naming for operating system (OS) and architecture combinations
```

> that can be used by all systems that read/write manifest files.

Reviewer: ```suggestion
that can be used by all systems that read and/or write manifest files.
```

As such, the following table should be considered the authoritative list:

+---------------+---------------+
| OS            | Tuple Name    |
+---------------+---------------+
| Linux         | linux         |

> | MacOS         | macos         |

Reviewer: ```suggestion
| macOS         | macos         |
```

> | Apple         | macos         |

Reviewer: ```suggestion
```

| Windows       | windows       |
| FreeBSD       | freebsd       |
| OpenBSD       | openbsd       |
+---------------+---------------+

+---------------+---------------+
| Architecture  | Tuple Name    |
+---------------+---------------+
| i386          | x86           |
| x86           | x86           |
| x86-64        | amd64         |
| x64           | amd64         |
| amd64         | amd64         |
| arm (32-bit)  | arm           |
| armbe (32-bit)| armbe         |
| arm64be       | arm64be       |
| aarch64       | arm64         |
| arm64         | arm64         |
| s390x         | s390x         |
| ppc           | powerpc       |
| ppc64         | powerpc64     |
| ppc64le       | powerpc64le   |
| riscv         | riscv         |
| riscv64       | riscv64       |
| sparc         | sparc         |
| sparc64       | sparc64       |

> | wasm          | wasm          |

Author: `wasm64` I believe

> | wasm          | wasm          |

Reviewer: It seems good to cover both wasm32 and wasm64 since wasm64 seems like a thing that will happen.

Would `unknown_wasm32`, and `unknown_wasm64` work?

> | wasm          | wasm          |

Author: Sure, I'll update this to cover both wasm32 and wasm64

+---------------+---------------+

> The construction of the platform tuple is: ``<os>_<arch>``, for example: ``linux_amd64``.

Reviewer: Fine as is but it's just slightly more clear if these names match the table headers (or we change the table headers to match these names).

```suggestion
The construction of the platform tuple is: ``<OS>_<Architecture>``, for example: ``linux_amd64``.
```

.. note::

>    For alternative scenarios such as using MUSL instead of glibc or MINGw, the tuple should

Reviewer: ```suggestion
   For alternative scenarios such as using musl instead of the GNU C Library (glibc) or MinGW, the tuple should
```

+---------------+-----------------+
| x86-64        | ``amd64``       |
+---------------+-----------------+
| x64           | ``amd64``       |
+---------------+-----------------+
| amd64         | ``amd64``       |
+---------------+-----------------+
| arm (32-bit)  | ``arm``         |
+---------------+-----------------+
| armbe (32-bit)| ``armbe``       |
+---------------+-----------------+
| arm64be       | ``arm64be``     |
+---------------+-----------------+
| aarch64       | ``arm64``       |
+---------------+-----------------+
| arm64         | ``arm64``       |
+---------------+-----------------+
| s390x         | ``s390x``       |
+---------------+-----------------+
| ppc           | ``powerpc``     |
+---------------+-----------------+
| ppc64         | ``powerpc64``   |
+---------------+-----------------+
| ppc64le       | ``powerpc64le`` |
+---------------+-----------------+
| riscv         | ``riscv``       |
+---------------+-----------------+
| riscv64       | ``riscv64``     |
+---------------+-----------------+
| sparc         | ``sparc``       |
+---------------+-----------------+
| sparc64       | ``sparc64``     |
+---------------+-----------------+

> | WASM (32-bit) | ``wasm32``      |

Reviewer: ```suggestion
| Wasm (32-bit) | ``wasm32``      |
```
